google/cloud/baremetalsolution/doc/main.dox

```diff
@@ -0,0 +1,99 @@
+/*!
+
+@mainpage Bare Metal Solution API C++ Client Library
+
+An idiomatic C++ client library for the [Bare Metal Solution API][cloud-service-docs],
+a service that provides ways to manage Bare Metal Solution hardware installed in
+a regional extension located near a Google Cloud data center.
+
+This library is **experimental**. Its APIs are subject to change without notice.
+
+This library requires a C++11 compiler. It is supported (and tested) on multiple
+Linux distributions, as well as Windows and macOS. The [README][github-readme]
+on [GitHub][github-link] provides detailed instructions to install the necessary
+dependencies, as well as how to compile the client library.
+
+### Setting up your repo
+
+In order to use the Bare Metal Solution API C++ client library from your own code,
+you'll need to configure your build system to discover and compile the Cloud
+C++ client libraries. In some cases your build system or package manager may
+need to download the libraries too. The Cloud C++ client libraries natively
+support [Bazel](https://bazel.build/) and [CMake](https://cmake.org/) as build
+systems. We've created a minimal, "Hello World", [quickstart][github-quickstart]
+that includes detailed instructions on how to compile the library for use in
+your application. You can fetch the source from [GitHub][github-link] as normal:
+
+@code{.sh}
+git clone https://github.com/googleapis/google-cloud-cpp.git
+cd google-cloud-cpp/google/cloud/baremetalsolution/quickstart
+@endcode
+
+@par Example: Quickstart
+
+The following shows the code that you'll run in the
+`google/cloud/baremetalsolution/quickstart/` directory,
+which should give you a taste of the Bare Metal Solution API C++ client library API.
+
+@include quickstart.cc
+
+## Environment Variables
+
+<!-- inject-endpoint-env-vars-start -->
+
+- `GOOGLE_CLOUD_CPP_BARE_METAL_SOLUTION_ENDPOINT=...` changes the default endpoint
+  (baremetalsolution.googleapis.com) used by `BareMetalSolutionConnection`.
+
+<!-- inject-endpoint-env-vars-end -->
```

Comment:
Q. Should we be leaving the "inject-endpoint-env-vars" markers in the output?

Reply:
yes, so the operation is idempotent

Reply:
Then I guess we best go and add them to all the existing `main.dox` files.